### PR TITLE
Add photo station backend

### PR DESCRIPTION
**Pull request by noobydp**
## Summary

This adds the backend foundation for reusable photo stations. A photo station stores a known camera/paper setup so later captures from the same setup can reuse or compare against saved paper corners instead of starting from scratch each time.

The PR includes:
- photo station request/response schemas
- a JSON-backed photo station store
- endpoints for listing, creating, updating, deleting, and reusing stations
- station suggestions for an uploaded session
- corner redetection support
- saving a station directly while confirming corners
- focused backend tests for the new station behavior

## Implementation Notes

The station store follows the existing local JSON-store pattern used elsewhere in the backend. Station records include paper size, image dimensions, source image path, saved corners, and timestamps.

The corner confirmation endpoint can now optionally receive `save_station_name`. When present, the backend creates a station from the confirmed session image and corners as part of the same workflow.

This also addresses review feedback from the original larger PR:
- Uses timezone-aware UTC timestamps instead of `datetime.utcnow()`.
- Centralizes station creation logic in a helper instead of duplicating it between routes.
- Copies the original uploaded image into station storage before unlinking the upload source.
- Adds backend coverage for station creation, suggestion behavior, invalid input, corrupt store handling, and the copy-before-unlink path.

## Tests

- `python -m pytest backend/tests/test_photo_stations.py`

Code and PR drafted with Codex

### CODE REVIEW

**Comment by jasonmadigan** [comment]
@noobydp I will get to this and your other related PRs at some stage over the next few days

**Comment by noobydp** [comment]
No problem, whenever you have time.
Thanks

File: backend/app/api/routes.py [hunk 1]
```diff
@@ -512,24 +720,57 @@ async def upload_image(request: Request, image: UploadFile, user_id: str = Depen
...
 
-    corners = image_processor.detect_paper_corners(str(image_path))
-    corner_points = [Point(x=c[0], y=c[1]) for c in corners] if corners else None
+    paper_size = None
+    applied_station_id = None
+    if station:
+        width_delta = _dimension_delta_percent(station.image_width, image_width)
+        height_delta = _dimension_delta_percent(station.image_height, image_height)
+        if max(width_delta, height_delta) > MAX_STATION_DIMENSION_DELTA_PERCENT:
+            raise HTTPException(status_code=400, detail="photo station image size differs too much from this upload")
```
Comment (jasonmadigan) [inline backend/app/api/routes.py:742]: **Important:** the upload is already written to disk at this point and no session record exists yet, so `uploads/{session_id}{ext}` is orphaned when this raises; each retry leaks a new file (fresh uuid per attempt). Check dims before writing, or unlink on raise. Also untested: the only station upload test crops to exact station dims, and `test_reuse_station_rejects_far_dimension_match` hits the duplicate raise in `reuse_photo_station_corners`, not this one.

File: backend/app/api/routes.py [hunk 2]
```diff
@@ -184,6 +257,135 @@ def _abs(rel_path: str | None) -> str | None:
...
+
+
+def _image_dimensions(content: bytes) -> tuple[int, int]:
+    img = Image.open(io.BytesIO(content))
+    return img.size
+
+
+def _session_image_dimensions(session: Session) -> tuple[int, int]:
+    if not session.original_image_width or not session.original_image_height:
+        raise HTTPException(status_code=400, detail="session has no upload dimensions")
```
Comment (jasonmadigan) [inline backend/app/api/routes.py:291]: **Important:** sessions persisted before this PR have no `original_image_width`, so `/sessions/{id}/station-suggestions` returns 400 for every pre-existing session once a station exists. Degrade to empty suggestions for sessions without dims instead.

File: backend/app/models/schemas.py [hunk 3]
```diff
@@ -215,6 +246,10 @@ class Session(BaseModel):
     tags: list[str] = []
     created_at: str | None = None
     original_image_path: str | None = None
+    original_image_width: int | None = None
+    original_image_height: int | None = None
+    capture_crop: CaptureCrop | None = None
+    station_image_path: str | None = None
```
Comment (jasonmadigan) [inline backend/app/models/schemas.py]: **Important:** nothing in production sets `station_image_path` (only a test, and #136/#137 don't wire it either), so the `_create_photo_station` fallback, `_station_image_referenced` and the `delete_session` branch are unreachable. Wire it or drop the field and helper. If wired, add a test proving a referenced station image survives session delete; current tests only prove deletion.

File: backend/tests/test_photo_stations.py [hunk 4]
```diff
@@ -0,0 +1,607 @@
...
+        "corners": [p.model_dump() for p in _corners()],
+        "paper_size": "a4",
+    })
+
+    assert resp.status_code == 200
+    assert resp.json()["station"] is None
+    assert not (tmp_path / "default" / "station-photos").exists()
+
+
+def test_set_corners_with_station_saves_station_photo(tmp_path, monkeypatch):
```
Comment (jasonmadigan) [inline backend/tests/test_photo_stations.py:630]: **Important:** success path only. Add a case where `shutil.copy2` raises inside `_copy_station_image` and assert the original upload survives; the unlink runs after `_create_photo_station`, and nothing currently proves a copy failure leaves the source intact.

File: backend/app/services/photo_station_store.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                            dirty = True
+                    self._stations[sid] = station
+                if dirty:
+                    self._save()
+            except OSError:
+                logger.error(f"Failed to load {self.file_path}: permission denied")
+                raise
+            except Exception as e:
+                logger.error(f"Failed to load {self.file_path}: {e}")
+                self._stations = {}
```
Comment (jasonmadigan) [inline backend/app/services/photo_station_store.py:36]: **Nit:** a corrupt `photo-stations.json` is reset to empty here and then permanently overwritten on the next save. Matches the session store pattern, but stations are long-lived; consider renaming the corrupt file aside before continuing.

File: backend/app/services/photo_station_store.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    def _load(self):
+        if self.file_path.exists():
+            try:
+                data = json.loads(self.file_path.read_text())
+                dirty = False
+                for sid, sdata in data.items():
+                    station = PhotoStation.model_validate(sdata)
+                    if station.image_path:
+                        normalized = station.image_path.replace("\\", "/")
```
Comment (jasonmadigan) [inline backend/app/services/photo_station_store.py]: **Nit:** this migration normalises data no release ever wrote. Drop it, or add a comment saying why it exists.

File: backend/app/api/routes.py [hunk 7]
```diff
@@ -512,24 +720,57 @@ async def upload_image(request: Request, image: UploadFile, user_id: str = Depen
...
+    if station_id:
+        station = get_photo_station_store(user_id).get(station_id)
+        if not station:
+            raise HTTPException(status_code=404, detail="photo station not found")
+
+    requested_crop = _parse_capture_crop(capture_crop)
+    applied_crop = requested_crop if requested_crop is not None else station.capture_crop if station else None
+
     content, ext, _ = ingest_image(content, ext, MAX_UPLOAD_DIM)
+    content, ext = _crop_image(content, ext, applied_crop)
```
Comment (jasonmadigan) [inline backend/app/api/routes.py]: **Nit:** the crop runs after the 2048px downscale, so a half-frame capture crop traces at roughly 1024px. Cropping first (after EXIF handling) preserves resolution.

File: backend/app/api/routes.py [hunk 8]
```diff
@@ -570,6 +822,142 @@ async def set_corners(request: Request, session_id: str, req: CornersRequest, us
...
+    if not session or not session.original_image_path:
+        raise HTTPException(status_code=404, detail="session original image not found")
+
+    detected = image_processor.detect_paper_corners(_abs(session.original_image_path))
+    if not detected:
+        raise HTTPException(status_code=422, detail="paper corners not detected")
+
+    corners = [Point(x=c[0], y=c[1]) for c in detected]
+    session.corners = corners
+    session.paper_size = session.paper_size or "a4"
```
Comment (jasonmadigan) [inline backend/app/api/routes.py]: **Nit:** silently defaults `paper_size` to a4.

File: backend/app/api/routes.py [hunk 9]
```diff
@@ -184,6 +257,135 @@ def _abs(rel_path: str | None) -> str | None:
...
+
+def _dimension_delta_percent(station_value: int, session_value: int) -> float:
+    if station_value <= 0:
+        return 100.0
+    return abs(session_value - station_value) / station_value * 100.0
+
+
+def _scaled_station_corners(station: PhotoStation, image_width: int, image_height: int) -> list[Point]:
+    sx = image_width / station.image_width
+    sy = image_height / station.image_height
```
Comment (jasonmadigan) [inline backend/app/api/routes.py:334]: **Nit:** worth a suggestions test with differing image heights so `sy` is actually asserted; current cases use y=0 or equal heights, and this codebase has form on y-axis bugs.

**Review by jasonmadigan — CHANGES_REQUESTED** [review]
thanks noobydp

**Comment by noobydp** [comment]
Updated this branch after review and rebased it onto current `main`.

Changes made in this update:
- Added the `PHOTO_STATIONS` feature flag and kept station endpoints dark unless enabled.
- Moved station upload dimension validation before writing the upload file, so rejected station uploads no longer leave orphaned files.
- Changed capture crop handling so cropping happens inside ingestion before the max-size downscale, preserving more source resolution.
- Made legacy sessions without stored image dimensions degrade to no station suggestions instead of returning 400.
- Dropped the unused `station_image_path` session field and the unreachable station-image reference cleanup path.
- Added finite-number validation for `Point` and `CaptureCrop` values so NaN cannot persist into station data.
- Simplified the station-image migration and now renames corrupt `photo-stations.json` aside before continuing with an empty store.
- Added tests for the above, including copy-failure source preservation, y-axis scaling, stale dimension rejection, feature flag behavior, and corrupt station storage handling.

Local validation:
- `python -m ruff check backend`
- `python -m pytest tests/test_photo_stations.py`
- `python -m pytest -k "not TestPermissionError"` on Windows: 318 passed, 8 deselected. The deselected cases are the existing Windows-only temp-file permission fixture issue; the full suite otherwise passed.

Code and PR drafted with Codex

**Comment by jasonmadigan** [comment]
thanks @noobydp!
